torch/_inductor/fx_passes/quantization.py

```diff
@@ -442,10 +442,55 @@ def _register_quantization_maxpool2d():
         )
 
 
+def _register_quantized_cat_lowering(
+    pattern,
+    computation_op,
+):
+    @register_lowering_pattern(
+        pattern,
+    )
+    def qcat(match: Match, inputs, dim, **kwargs):
+        # Refer to the implementation
+        # https://github.com/pytorch/pytorch/blob/416bf4e3e7c5f90bbc306e55501881b4f8bde981/torch/_inductor/fx_passes/post_grad.py#L234
+        # the bias, mat1, mat2 are kept order in the inputs.
+        uint8_inputs = [input[0] for input in inputs]
+        return L[computation_op](uint8_inputs, dim)
+
+    return qcat
+
+
+_raw_dequantize_per_tensor_activation_pattern = CallFunction(
+    aten.mul.Tensor,
+    CallFunction(
+        aten.sub.Tensor,
+        CallFunction(
+            prims.convert_element_type.default,
+            Arg(),
+            Arg(),
+        ),
+        Arg(),
+    ),
+    Arg(),
+)
+
+
+def _register_quantization_cat():
+    dequantize_cat_pattern = CallFunction(
+        aten.cat.default,
+        ListOf(_raw_dequantize_per_tensor_activation_pattern),
```

Comment:
The rewrite is only valid when the dequantization parameters are the same for all the input tensors to concat? Should we check it?

Reply:
Thanks for the comment. Yean, we need to check the quantization parameters and add it. Please help to take a look again.